lib/rucio/client/uploadclient.py

```diff
@@ -500,7 +530,7 @@ def _collect_file_info(self, filepath, item):
 
         return new_item
 
-    def _collect_and_validate_file_info(self, items):
+    def _collect_and_validate_file_info(self, items: "Iterable[FileToUploadDict]") -> list["FileToUploadWithCollectedInfoDict"]:
```

Comment:
Why list and not Sequence?

Reply:
The general type annotation convention is to use abstract collection types e.g. `Iterable`, `Sequence` for function arguments, but to use specific types like `list` for the return type: https://docs.python.org/3/library/typing.html#typing.List